pypeit/calibrations.py

```diff
@@ -371,7 +377,12 @@ def get_arc(self):
 
         # If a processed calibration frame exists and we want to reuse it, do
         # so:
-        if cal_file.exists() and self.reuse_calibs:
+        if force == 'remake':
+            pass
+        elif force == 'reload' and not cal_file.exists():
+            self.success = False
+            return
+        elif force == 'reload' or (self.reuse_calibs and cal_file.exists()): 
```

Comment:
I recommend abstracting this a bit.  I'd add this member function:
```
def process_load_selection(self, frame, cal_file, force):
    """
    Process how pypeit should use any pre-existing calibration files.

    If loading is requested but the calibration file (``cal_file``) does
    not exist, ``self.success`` is set to False, and None is returned.

    Args:
        frame (:obj:`dict`):
            A dictionary with two elements: ``type`` is the string
            defining the frame type and ``class`` is the pypeit class
            used to load the pre-existing calibration file.
        cal_file (:obj:`str`, `Path`_):
            Path to the calibration file.
        force (:obj:`str`):
            Defines how to treat a pre-existing calibration file.  Must
            be one of the following options:
                
                - ``'remake'``: Force the calibration be remade.

                - ``'reload'``: Reload the frame if it exists.

                - ``None``: Load the existing frame if it exists and
                  ``self.reuse_calibs=True``.

    Returns:
        :obj:`object`:  Either the loaded calibration object or None.
    """
    if force not in [None, 'remake', 'reload']:
        msgs.error(f'`force` keyword must be None, remake, or reload, not {force}')
    if force == 'remake':
        return None
    _cal_file = Path(cal_file).absolute()
    if force == 'reload' and not _cal_file.exists():
        msgs.warn(f"{_cal_file} does not exist; cannot reload "
                  f"{frame['class'].__name__} calibration.")
        self.success = False
        return None
    if force == 'reload' or (self.reuse_calibs and _cal_file.exists()): 
        return = frame['class'].from_file(_cal_file, chk_version=self.chk_version)
```
And then the lines within each `get_*` function would replace this block with:
```
self.?? = self.process_load_selection(frame, cal_file, force)
if not self.success or self.?? is not None:
    return self.??
```
where `??` is `msarc` in this function.